parts/inc/uv

```diff
@@ -103,14 +103,23 @@ my_strnlen(const char *str, Size_t maxlen)
  * intended to work with modern perls, the tighter restrictions could be
  * relaxed.  khw thinks this is unlikely, but has been wrong in the past. */
 
+#if { VERSION < 5.10.0 }
+# define cU8 U8
+#else
+# define cU8 const U8
+#endif
+
+
 #ifndef utf8_to_uvchr_buf
    /* Choose which underlying implementation to use.  At least one must be
     * present or the perl is too early to handle this function */
 #  if defined(utf8n_to_uvchr) || defined(utf8_to_uv)
 #    if defined(utf8n_to_uvchr)   /* This is the preferred implementation */
 #      define _ppport_utf8_to_uvchr_buf_callee utf8n_to_uvchr
-#    else
+#    elif { VERSION >= 5.6.1 }
 #      define _ppport_utf8_to_uvchr_buf_callee utf8_to_uv
+#    else
+#      define _ppport_utf8_to_uvchr_buf_callee(s, curlen, retlen, flags) utf8_to_uv(s, retlen)
 #    endif
```

Comment:
This is really incompatible. Two-argument version of `utf8_to_uv` expects second argument `retlen` as `I32 *`, but `_ppport_utf8_to_uvchr_buf_callee`'s `retlen` is of type `STRLEN *`.

So I rather suggest following change:
```c
...
#    elif { VERSION >= 5.6.1 }
#      define _ppport_utf8_to_uvchr_buf_callee utf8_to_uv
#    endif
```
I.e. do not define `_ppport_utf8_to_uvchr_buf_callee` at all.

Reply:
👍